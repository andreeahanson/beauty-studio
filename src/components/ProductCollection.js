import React from 'react';
import ProductCard from '../containers/ProductCard';
import PropTypes from 'prop-types';

const ProductCollection = (products) => {
  const allProducts = products.products.map(product => {
    if(product.price != 0){
    return (
    <ProductCard 
    id={product.created_at}
    key={product.created_at}
    brand={product.brand}
    name={product.name} 
    description={product.description}
    price={product.price} 
    image={product.image_link} 
    colors={product.product_colors}
    type={product.product_type}
    tagList={product.tag_list}
    />)
    }
  })
  return allProducts
}

export default ProductCollection;

ProductCollection.propTypes = {
  products: PropTypes.array
}  