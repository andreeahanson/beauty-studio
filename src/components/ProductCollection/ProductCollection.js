import React from 'react';
import ProductCard from '../ProductCard/ProductCard';
import './ProductCollection.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export const ProductCollection = ({products, toggleFavorite, isLoading}) => {
  const allProducts = products.map(product => {
    if(product.price != 0){
    return (
    <ProductCard 
    id={product.id}
    key={product.id}
    brand={product.brand}
    name={product.name} 
    description={product.description}
    price={product.price} 
    image={product.image_link} 
    colors={product.product_colors}
    type={product.product_type}
    tagList={product.tag_list}
    toggleFavorite={toggleFavorite}
    />)
    }
  })
  if(!allProducts.length && !isLoading){
    return (<h2 className="pink">No loves added yet</h2>)
  } else {
    return allProducts
  }
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading
})

export default connect(mapStateToProps)(ProductCollection);

ProductCollection.propTypes = {
  products: PropTypes.array,
  toggleFavorite: PropTypes.func,
  isLoading: PropTypes.bool
}  