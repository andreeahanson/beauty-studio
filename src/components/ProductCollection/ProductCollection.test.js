import React from 'react';
import ProductCollection from './ProductCollection';
import { shallow } from 'enzyme';

describe('ProductCollection', () => {
  let products = [{id:1}, {id:2}]
  let mockToggleFavorite = jest.fn()


  it('should match the snapshot', () => {
    let wrapper=shallow(
    <ProductCollection 
    products={products}
    toggleFavorite={mockToggleFavorite}
    />)

    expect(wrapper).toMatchSnapshot();
  })
}) 