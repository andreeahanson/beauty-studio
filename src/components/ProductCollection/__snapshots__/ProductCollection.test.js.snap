// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCollection should match the snapshot 1`] = `ShallowWrapper {}`;
